src/States/GettingReadyState.js

```diff
@@ -130,6 +143,13 @@ function GettingReady() {
       friendDialogue.destroy();
     }
 
+    if (progress >= dialogueTree.length) {  //@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@ ISSUE!! @@@@@@@@@@@ -> progressing after loop
+      transitionToNextState();
+      return;
```

Comment:
Ah, this one's my bad. I know I did this and I can submit a fix later, but putting `return` calls in the middle of functions tends to be bad practice.